R/remote-info.R

```diff
@@ -0,0 +1,43 @@
+# This code is difficult to read and has caused problems in the past. Heed my
+# warning. This so that as.data.frame() is given a list. The *_subdir fields are
+# missing in most cases. If they were included in the main list() calls, list()
+# would include a field with that name with a NULL value.
+#
+# Creating a list and then concatenating the possibly-NULL subdir fields means
+# that they are NULL, they will not appear at all in the resulting list at all.
+# The resulting data frame is later appended to the DESCRIPTION file, so this is
+# desirable.
+getRemoteInfo <- function(pkgRecord) {
```

Comment:
The old GitHub handling had a very simple (to my eye) shape:

https://github.com/rstudio/packrat/blob/d152f12602e2595e3cfe3c73acc9a1a2de3ecad0/R/restore.R#L307-L315

Using that as inspiration, that could give you:

```r
return(as.data.frame(as.list(c(
             RemoteType     = pkgRecord$source,
             GithubRepo     = pkgRecord$gh_repo,
             GithubUsername = pkgRecord$gh_username,
             GithubRef      = pkgRecord$gh_ref,
             GithubSHA1     = pkgRecord$gh_sha1,
             GithubSubdir = pkgRecord$gh_subdir
)), stringsAsFactors = FALSE)
```

Reply:
It's almost certainly not the most obvious way to materialize it — this was the original way it was materialized, and that same impulse led me to try to simplify it and break it.

I might have another go at simplifying — and just do it in a more verbose, obvious way — now that we have the tests.

Reply:
Maybe I didn't try `as.list`, just `list`.